network/ovs_endpoint_snatroute_linux.go

```diff
@@ -8,8 +8,7 @@ import (
 )
 
 func NewSnatClient(client *OVSEndpointClient, snatBridgeIP string, localIP string, epInfo *EndpointInfo) {
-	if client.enableSnatOnHost || client.allowInboundFromHostToNC {
-
+	if client.enableSnatOnHost || client.allowInboundFromHostToNC || client.enableSnatForDns {
```

Comment:
can you add client.allowInboundFromNCToHost also? looks like its missed somehow by my previous PR